lnwallet/chancloser/chancloser.go

```diff
@@ -822,24 +837,20 @@ func (c *ChanCloser) ProcessCloseMsg(msg lnwire.Message) ([]lnwire.Message,
 		// negotiation.
 		c.state = closeFinished
 		matchingOffer := c.priorFeeOffers[remoteProposedFee]
-		return []lnwire.Message{matchingOffer}, true, nil
+
+		return fn.Some(*matchingOffer), nil
 
 	// If we received a message while in the closeFinished state, then this
-	// should only be the remote party echoing the last ClosingSigned message
-	// that we agreed on.
+	// should only be the remote party echoing the last ClosingSigned
+	// message that we agreed on.
 	case closeFinished:
-		if _, ok := msg.(*lnwire.ClosingSigned); !ok {
-			return nil, false, fmt.Errorf("expected lnwire.ClosingSigned, "+
-				"instead have %v", spew.Sdump(msg))
-		}
 
-		// There's no more to do as both sides should have already broadcast
-		// the closing transaction at this state.
-		return nil, true, nil
+		// There's no more to do as both sides should have already
+		// broadcast the closing transaction at this state.
+		return noClosing, nil
 
-	// Otherwise, we're in an unknown state, and can't proceed.
 	default:
```

Comment:
I think we need to cache the peer's `closing_signed` because if they send us a `revoke_and_ack` followed by `closing_signed`, we might call `ProcessClosingSigned` before the link calls `BeginNegotiation`. Then we'd hit this default case and error out